app/controllers/Application.scala

```diff
@@ -309,29 +312,29 @@ object Application extends Controller {
     } yield j.asInstanceOf[JsObject]).toOption
 
     val fpath = nbm.newNotebook(
-                    path,
-                    customLocalRepo orElse config.localRepo,
-                    customRepos orElse config.repos,
-                    customDeps orElse config.deps,
-                    customImports orElse config.imports,
-                    customMetadata orElse config.sparkConf)
+      path,
+      customLocalRepo orElse config.localRepo,
+      customRepos orElse config.repos,
+      customDeps orElse config.deps,
+      customImports orElse config.imports,
+      customMetadata orElse config.sparkConf)
     Try(Redirect(routes.Application.contents("notebook", fpath)))
   }
 
-  def copyingNb(fp:String) = {
-    val fromPath = URLDecoder.decode(fp)
+  def copyingNb(fp: String) = {
+    val fromPath = URLDecoder.decode(fp, UTF_8)
     Logger.info("Copying notebook:" + fromPath)
     val np = nbm.copyNotebook(fromPath)
     Try(Ok(Json.obj("path" → np)))
   }
 
-  def newNotebook(path:String, tryJson:Try[JsValue]) = {
-    def findkey[T](x:JsValue, k:String)(init:T)(implicit m:ClassTag[T]):Try[T] =
-      (x \ k) match {
-        case j:JsUndefined => Failure(new IllegalArgumentException("No " + k))
+  def newNotebook(path: String, tryJson: Try[JsValue]) = {
+    def findkey[T](x: JsValue, k: String)(init: T)(implicit m: ClassTag[T]): Try[T] =
+      x \ k match {
```

Comment:
I'd rather prefer keeping the enclosing parenthesis, it makes the `match`'s scope more clearer IMHO